Doughnut/Player/Player.swift

```diff
@@ -182,7 +192,7 @@ final class Player: NSObject {
           episode.playPosition = Int(time.seconds)
           episode.duration = Int(avPlayer.currentItem?.asset.duration.seconds ?? 0)
 
-          if episode.duration > 0, (Double(episode.playPosition) / Double(episode.duration)) > Self.playedThreshold {
+          if episode.duration > 0, (Double(episode.playPosition) / Double(episode.duration)) > self.playedThreshold {
```

Comment:
Update: seems `AVPlayer` always fires the timeObserver when played to the end and there's no rounding issues happening. Just changing `>` to `>=` would be enough.
<img width="894" alt="2022-10-12 20 55 15" src="https://user-images.githubusercontent.com/8158163/195348138-29179db2-de4c-44b9-a8d4-d84689a530a8.png">